test/test_nn.py

```diff
@@ -8671,6 +8671,35 @@ def test_pdist_cuda_gradgrad_unimplemented(self):
         inp = torch.randn(4, 5, device='cuda', requires_grad=True)
         gradgradcheck(F.pdist, (inp,))
 
+    def test_binary_cross_entropy_grads(self):
+        import torch.nn.functional as F
+        for device in device_():
+            input = torch.rand(3, 3, dtype=torch.double, device=device, requires_grad=True)
+            target = torch.rand(3, 3, dtype=torch.double, device=device)
+
+            gradcheck(F.binary_cross_entropy, [input, target])
+            gradgradcheck(F.binary_cross_entropy, [input, target])
+
+            # now with diffentiable target
+            target.requires_grad_(True)
+            # grad wrt target is not a native function (in a sense that it is
+            # implemented on top of "high-level" functions, and does not have
+            # explicit device-specific dispatches)
+            # and uses in-place operations that trigger vmap failures in gradcheck,
+            # hence the flag `check_batched_grad=False`
+            # TODO: use native kernels to circumvent this limitation.
+            gradcheck(F.binary_cross_entropy, [input, target], check_batched_grad=False)
+            with self.assertRaisesRegex(RuntimeError, "not implemented"):
+                # binary_cross_entropy_backward only returns grads wrt input,
```

Comment:
I'm not sure to understand what this comment is trying to explain.
Is it just: "we don't support double backward wrt to target" ?

We'll only need to update `binary_cross_entropy_target_backward` to be differentiable (by removing inplace conditioned on the current GradMode) and add a formula for target: for `binary_cross_entropy_backward`

Reply:
but `binary_cross_entropy_backward` returns `grad_wrt_input` and does not handle `grad_wrt_target`, would that still work?

Reply:
You can have the backward split into two part: one function that compute grads wrt to input and another that compute grads wrt to target.
But if both of these use target, then the double backward will need the double backward of both to compute the double gradient for target.

That will work yes, if you think of this as a single backward, it's the same as if you're doing `res = f1(inp) + f2(inp)` and ask for gradients wrt `inp`.

Reply:
If I decide to split it into two functions, would it be possible to access grad_target and grad_input in both double backwards for input and target?